bids/reports/report.py

```diff
@@ -0,0 +1,115 @@
+"""Generate publication-quality data acquisition methods section from BIDS dataset.
+"""
+from __future__ import print_function
+import json
+from os.path import dirname
+from os.path import abspath
+from os.path import join as pathjoin
+from collections import Counter
+
+from bids.reports import utils
+
+
+class BIDSReport(object):
+    """
+    Generates publication-quality data acquisition methods section from BIDS
+    dataset.
+
+    Parameters
+    ----------
+    layout : :obj:`bids.grabbids.BIDSLayout`
+        Layout object for a BIDS dataset.
+    config : :obj:`str` or :obj:`dict`, optional
+        Configuration info for methods generation. Can be a path to a file
+        (str), a dictionary, or None. If None, loads and uses default
+        configuration information.
+    """
+    def __init__(self, layout, config=None):
+        self.layout = layout
+        if config is None:
+            root = dirname(abspath(__file__))
+            config = pathjoin(root, 'config', 'converters.json')
+            with open(config) as fobj:
+                config = json.load(fobj)
+
+        if not isinstance(config, dict):
+            raise ValueError('Input config must be None, dict, or path to json '
+                             'file containing dict.')
+
+        self.config = config
+
+    def generate(self, task_converter=None):
```

Comment:
Is there a reason to pass the `task_converter` here, but all the other converters in the initializer? I get that tasks might vary much more on a project-by-project basis than most of the other fields, but since the user has to pass a single `BIDSLayout` to the init, the expectation is presumably that any `generate` calls are always going to apply to the same project. So it might make sense to move `task_converter` to the initializer as well--unless you think it will be a common use case for someone to only want to do one task out of many, and specify the task dict on the fly instead of creating one for the project as a whole.

Reply:
I was thinking it might be good just to use the `TaskName` field in the metadata, when available. Presumably, that would be the same full name as the one users would pass in as `task_converter`. Otherwise, I can move it back into `__init__` and just merge `task_converter` into `config` at the beginning.